### DISCUSSION TITLE
Wireless update failed with Input/output error

### DISCUSSION THREAD

**Discussion by lucaspereira512** [emporia-vue-local/esphome#16] [Q&A]
Ive been trying to update my emporia vue 2 running esp home wirelessly, and im getting the following error:

```
INFO Reading configuration /config/esphome/emporia-sensor.yaml...
INFO Detected timezone 'America/New_York'
INFO Generating C++ source...
INFO Compiling app...
Processing emporiavue2 (board: esp32dev; framework: espidf; platform: platformio/espressif32 @ 3.5.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - framework-espidf @ 3.40302.0 (4.3.2) 
 - tool-cmake @ 3.16.4 
 - tool-ninja @ 1.7.1 
 - toolchain-riscv32-esp @ 8.4.0+2021r2-patch2 
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch2 
 - toolchain-xtensa-esp32s2 @ 8.4.0+2021r2-patch2
Reading CMake configuration...
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
No dependencies
*** [/data/emporiavue2/.pioenvs/emporiavue2/src/esphome/components/api/api_connection.o] /data/cache/platformio/packages/framework-espidf/components/freertos/include/freertos/FreeRTOS.h: Input/output error
========================== [FAILED] Took 4.90 seconds ==========================
```
Doesnt seem to give me much information, or at least not enough for my mortal mind to figure out. Any guidance would be greatly appreciated.

I can upload my redacted configuration if that seems to be necessary.

Thanks

**Comment by flaviut** [comment]
Seems like it's some sort of caching issue. I'm not sure where you're running the update from, but this will be either deleting some files (do not delete your configuration!) or a button that you need to press.

**Reply by lucaspereira512** [reply]
So im running the update through ESPHome through Home Assistant. When you say delete files, which ones are you referring to?

**Reply by flaviut** [reply]
I don't have ESPHome installed through Home Assistant, but I found a screenshot of the button online:

![87b777643ac9c33c2d6e9b5e73522fa33898dd56](https://user-images.githubusercontent.com/5213469/186049019-12cfe3a8-a27e-4575-aa8b-f01c28160b35.jpeg)

Let me know how "Clean Build Files" works.

**Reply by lucaspereira512** [reply]
Hmmm, i never noticed that. However when I tried it, it still gave me the same error

**Reply by flaviut** [reply]
Is your drive full?

Do you see anything interesting in the system log, `dmesg`, when you run into this error?

Can you run whatever the appropriate disk corruption checking program is for your disk? It usually has something like `fsck` in the name.

**Reply by lucaspereira512** [reply]
It won't let me copy the output from the terminal, but here's an image of what the dmesg is outputting when it fails.

![image](https://user-images.githubusercontent.com/49882781/186192963-871eba23-9f0f-41e8-9aca-adbf0ba8fb28.png)

I also checked the disk, it seems healthy, and only using 66% of the disk

**Reply by flaviut** [reply]
Sorry :(

Looks like it's time to back everything up one last time and get a new disk. This one seems to be starting to go bad.

**Reply by lucaspereira512** [reply]
so my home assistant installation is running on a VM using vSphere 7.0, so id be very suprised if the only issue that would manifest is being unable to wirelessly push an update to an ESP device. ill keep looking around though. I may try a reinstall with the device physically plugged in, but its in my breaker box, and i dont feel like taking it apart right now.

Thanks though, when I get around to it, ill update the discussion.

**Reply by flaviut** [reply]
> so my home assistant installation is running on a VM using vSphere 7.0, so id be very suprised if the only issue that would manifest is being unable to wirelessly push an update to an ESP device.

The message there definitely indicates a disk error, but it might not be related to your issue, especially if it was more than 7 minutes after startup that you tried it (which is what the message indicates). I would imagine that an disk array would be more reliable than the SD cards some people use.

Anyway, good luck! This is definitely not an ESPHome issue though, and I'd be very surprised if a non-wireless flashing worked any better, especially since things are failing during compilation and not upload.
